File: MOBILE/achadinhos_online/src/interfaces/Venda.ts
import {Funcionario} from './Funcionario';

export interface Venda {
  id?: number;
  funcionario: Funcionario;
  venda: number;
  comissao?: number;
  selecionado?: boolean;
}

export interface VendaDTO {
  id?: number;
  idFuncionario: number;
  valor: number;
}

export interface TotalVenda {
  totalVenda: number;
  valorTotal: number;
  comissaoTotal?: number;
}

export interface StatusLojinha {
  funcionarios: number;
  produtos: number;
  vendas: number;
}

File: projeto_extra/API/crud-java/src/main/java/com/example/crudjava/domain/venda/Venda.java
package com.example.crudjava.domain.venda;

import com.example.crudjava.domain.funcionario.Funcionario;
import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Table(name = "vendas")
@Entity(name = "Venda")
@Getter
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(of = "id")
public class Venda {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "funcionario_id")
    private Funcionario funcionario;
    @Setter
    private String nomeCliente;
    @Setter
    private BigDecimal valorTotal;
    @Setter
    private BigDecimal comissaoTotal;
    @Column(name = "date_created", updatable = false)
    private LocalDateTime dateCreated;

    @Column(name = "date_updated")
    private LocalDateTime dateUpdated;

    public Venda(Funcionario funcionario, BigDecimal valorTotal, String nomeCliente) {
        this.funcionario = funcionario;
        this.valorTotal = valorTotal;
        this.nomeCliente = nomeCliente;
        this.comissaoTotal = calculaComissao(funcionario, valorTotal);
    }

    public BigDecimal calculaComissao(Funcionario funcionario, BigDecimal valor) {
        return valor.multiply(
                BigDecimal.valueOf(funcionario.getPorcentagem()).divide(BigDecimal.valueOf(100)));
    }
}
